src/Sylius/Bundle/ApiBundle/Tests/CommandHandler/ChangeAddressOrderHandlerTest.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Tests\CommandHandler;
+
+use Doctrine\Common\DataFixtures\Purger\ORMPurger;
+use Doctrine\Persistence\ObjectManager;
+use Fidry\AliceDataFixtures\LoaderInterface;
+use Fidry\AliceDataFixtures\Persistence\PurgeMode;
+use SM\Factory\FactoryInterface as StateMachineFactoryInterface;
+use Sylius\Bundle\ApiBundle\Command\Checkout\AddressOrder;
+use Sylius\Bundle\ApiBundle\CommandHandler\Checkout\AddressOrderHandler;
+use Sylius\Bundle\ApiBundle\Mapper\AddressMapperInterface;
+use Sylius\Component\Addressing\Model\AddressInterface;
+use Sylius\Component\Core\Model\Address;
+use Sylius\Component\Core\Repository\CustomerRepositoryInterface;
+use Sylius\Component\Core\Repository\OrderRepositoryInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+
+final class ChangeAddressOrderHandlerTest extends KernelTestCase
+{
+    /**
+     * @test
+     */
+    public function it_changes_address_order_without_duplication_in_database(): void
+    {
+        $container = self::bootKernel()->getContainer();
```

Comment:
https://github.com/Sylius/Sylius/pull/12719#discussion_r656928647

Did you take it into consideration?